### ISSUE TITLE
Error parsing basic project file (using FSAC through Spacemacs)

### ISSUE THREAD

**Issue by halfbro** [enricosada/dotnet-proj-info#60]
I'm using FSAC with Spacemacs `lsp-mode` to try and develop on my Windows 10 machine. I've installed Visual Studio 2019 and Visual Studio Code (with Ionide), which both work.

I'm trying to get Spacemacs to parse this test project:
https://github.com/halfbro/TestFSACProject
However, `lsp-mode` reports that the library I included from paket is not available, with a `Namespace not defined` error at `open Microsoft.Azure.Search`

I ran FSAC with the `--verbose` flag, and this was the output from the `*fsac::stderr*` buffer:
```
[LSP call] Initialize
[LSP call] Initialized
[LSP call] TextDocumentDidOpen
[LSP] Notify - Workspace (WorkspaceLoad false)
[LSP] Notify - Workspace
  (ProjectLoading "c:\Users\m0l02c6\Test\TestFSACProject\TestDotNet.fsproj")
[LSP call] TextDocumentCodeAction
[BACKGROUND SERVICE] Msg: File update c:\Users\m0l02c6\Test\TestFSACProject\Program.fs
[BACKGROUND SERVICE] Msg: Files to check []
[LSP] Notify - Workspace
  (ProjectError
     (GenericError
        ("c:\Users\m0l02c6\Test\TestFSACProject\TestDotNet.fsproj",
         "MSBuild failed with exitCode 1 Working Directory: 'c:\Users\m0l02c6\Test\TestFSACProject' Exe Path: 'dotnet' Args: 'msbuild c:\Users\m0l02c6\Test\TestFSACProject\TestDotNet.fsproj /p:SkipCompilerExecution=true /p:ProvideCommandLineArgs=true /p:CopyBuildOutputToOutputDirectory=false /p:UseCommonOutputDirectory=true /t:_Inspect_FscArgs /p:_Inspect_FscArgs_OutFile=C:\Users\m0l02c6\AppData\Local\Temp\tmpDBB4.tmp.FscArgs.txt /p:DesignTimeBuild=true /t:_Inspect_GetResolvedProjectReferences /p:_Inspect_GetResolvedProjectReferences_OutFile=C:\Users\m0l02c6\AppData\Local\Temp\tmpDBC5.tmp.GetResolvedProjectReferences.txt /t:_Inspect_GetProperties /p:_Inspect_GetProperties_OutFile=C:\Users\m0l02c6\AppData\Local\Temp\tmpDBC6.tmp.GetProperties.txt /t:_Inspect_Items /p:_Inspect_Items_OutFile=C:\Users\m0l02c6\AppData\Local\Temp\tmpDBE6.tmp.Items.txt /p:CustomAfterMicrosoftCommonTargets=C:\Users\m0l02c6\AppData\Local\Temp\tmpDBE7.tmp.proj-info.hook.targets /p:CustomAfterMicrosoftCommonCrossTargetingTargets=C:\Users\m0l02c6\AppData\Local\Temp\tmpDBE7.tmp.proj-info.hook.targets /nologo /verbosity:quiet'
Log: 
searching deprecated target file in 'c:\Users\m0l02c6\Test\TestFSACProject\obj\TestDotNet.fsproj.proj-info.targets'.
writing helper target file in 'C:\Users\m0l02c6\AppData\Local\Temp\tmpDBE7.tmp.proj-info.hook.targets'
System.ArgumentException: Item has already been added. Key in dictionary: 'EM_PARENT_PROCESS_ID'  Key being added: 'EM_PARENT_PROCESS_ID'
   at System.Collections.Hashtable.Insert(Object key, Object nvalue, Boolean add)
   at System.Environment.ToHashtable(IEnumerable`1 pairs)
   at System.Environment.GetEnvironmentVariables()
   at System.Diagnostics.ProcessStartInfo.get_Environment()
   at Microsoft.DotNet.Cli.Utils.ForwardingAppImplementation.GetProcessStartInfo()
   at Microsoft.DotNet.Cli.Utils.MSBuildForwardingAppWithoutLogging.GetProcessStartInfo()
   at Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingApp.GetProcessStartInfo()
   at Microsoft.DotNet.Tools.MSBuild.MSBuildCommand.Run(String[] args)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)

")))
[LSP] Notify - Workspace
  (ProjectError
     (GenericError
        ("c:\Users\m0l02c6\Test\TestFSACProject\TestDotNet.fsproj",
         "Project file 'c:\Users\m0l02c6\Test\TestFSACProject\TestDotNet.fsproj' parsing failed")))
[LSP] Notify - Workspace (WorkspaceLoad true)
[LSP call] TextDocumentDidOpen - workspace ready
[FSharpChecker] Current Queue Length: 0
[Checker] ParseAndCheckFileInProject - c:\Users\m0l02c6\Test\TestFSACProject\Program.fs
[LSP call] TextDocumentCodeAction
[LSP call] TextDocumentHover
[LSP call] TextDocumentSignatureHelp
[LSP] PositionHandler - Position request: c:\Users\m0l02c6\Test\TestFSACProject\Program.fs at (2,1)
[LSP] PositionHandler - Position request: c:\Users\m0l02c6\Test\TestFSACProject\Program.fs at (2,1)
[FSharpChecker] Current Queue Length: 0
[Checker] TryGetRecentCheckResultsForFile - c:\Users\m0l02c6\Test\TestFSACProject\Program.fs
[LSP] PositionHandler - Cached typecheck results not yet available
[Commands] TryGetLatestTypeCheckResultsForFile - c:\Users\m0l02c6\Test\TestFSACProject\Program.fs; State - Some 0; Checked - <null>
[Commands] TryGetLatestTypeCheckResultsForFile - c:\Users\m0l02c6\Test\TestFSACProject\Program.fs; State - Some 0; Checked - Some 0
[FSharpChecker] Current Queue Length: 0
[Checker] TryGetRecentCheckResultsForFile - c:\Users\m0l02c6\Test\TestFSACProject\Program.fs
[LSP] Notify - FileParsed "c:\Users\m0l02c6\Test\TestFSACProject\Program.fs"
[LSP] Notify - ParseError
  (Errors
     ([|c:\Users\m0l02c6\Test\TestFSACProject\Program.fs (4,16)-(4,21) typecheck error The namespace 'Azure' is not defined.|],
      "c:\Users\m0l02c6\Test\TestFSACProject\Program.fs"))
[LSP call] TextDocumentCodeAction
```

I did some digging and noticed that the code that was failing was coming from this repository, specifically [this line](https://github.com/fsharp/FsAutoComplete/blob/49056f4ca916f7ab9d8c40a77f1e515c86ae3a8e/src/FsAutoComplete.Core/Workspace.fs#L29) which calls into the project loader. I also noticed the strange issue with the (emacs-generated?) environment variable `EM_PARENT_PROCESS_ID`, which is probably what is causing the project loading to fail, but wanted to post this report anyway in case someone else has a solution/experience with `msbuild`.

So, to summarize:
1) `git clone` example repository
2) `dotnet build` in project directory
2) Start Spacemacs
3) Open the project directory in Spacemacs
4) Open `Program.fs`, install `fsautocomplete` according to `lsp-mode`'s F# tooling, and select the project directory as the project root
5 Expected result) Project is loaded successfully
5 Actual result) FSAC throws the above error, and I cannot use external libraries in projects

I can provide configuration files and versions if necessary, but I suspect the problem to be something else. I will also test this later tonight on my home machine (Windows 8) to try to reproduce, however I primarily develop on this machine, so it would be nice to figure out why it is not working.

**Comment by enricosada** [comment]
maybe is related to https://github.com/dotnet/cli/issues/9461 and https://github.com/dotnet/corefx/issues/30341

@halfbro what is the dotnet version (get with `dotnet --info`)

**Comment by halfbro** [comment]
dotnet sdk version is 2.2.300
dotnet version is 2.2.5

I tried on my home machine, was not able to get FSAC working in a reasonable time frame, but did notice that the variable was being set twice again, so maybe it isn't just me having this issue. I was able to install F# LSP mode on my Linux machine pretty successfully, so that's good at least.

I think that the error stems from Emacs setting the same environment variable twice, I'm going to try to workaround this issue by maybe clearing the value in Emacs during initialization

**Comment by halfbro** [comment]
I realized that spacemacs actually loads environment variables from `~/.spacemacs.env`, so I opened that file up to check if it contained the weird variable. Sure enough, in that file there were duplicates of `EM_PARENT_PROCESS_ID`, which I deleted (the text at the top of the file suggests it is safe to do so).

I got some null reference errors after that, but it's time for me to go to sleep for the night

**Comment by halfbro** [comment]
I'm going to close this issue as I think the error is from `~/.spacemacs.env`. I'm able to edit a different project without any errors after making the above change.
